### ISSUE TITLE
HTTP 403 forbidden

### ISSUE THREAD

**Issue by rathsky** [jgeboski/bitlbee-steam#94]
Getting this since about two days, plugin is at HEAD. It logs in fine, then after some five seconds it's consistently kicked. No more log other than that 403 code.

**Comment by jgeboski** [comment]
I am not able to recreate this issue.  Have you tried re-authenticating?  If it still fails, please provide the request which is failing via `BITLBEE_DEBUG_STEAM=1 /usr/sbin/bitlbee -Dnv`.

Re-authenticate:
```
account <acc> off
account <acc> set -del token
account <acc> on
```

**Comment by rathsky** [comment]
Same result after re-authentication. I'm guessing BITLBEE_DEBUG_STEAM prints to console, and not some file or the IRC client? 

Anyway, there are only three (or so) lines printed for each failed attempt (nothing else anywhere), all a repetition of

> (process:7791): GLib-CRITICAL **: Source ID 668 was not found when attempting to remove it

With the "Source ID" being a number increasing by an apparently random amount (like 8 or 209).

bitlbee reports itself is "3.4.1+20150811+master+17-g86fd261-git".

**Comment by rathsky** [comment]
Hm, well for whatever reason the 403 no longer occurs. Guess it might not have been related to bitlbee after all. I'll reopen if necessary.

**Comment by rathsky** [comment]
Getting it again. I recompiled bitlbee-steam with --enable-debug, started bitlbee with

> BITLBEE_DEBUG=1 BITLBEE_DEBUG_STEAM=1 sudo -u bitlbee bitlbee -Fn

No additional [debug] messages at all.

**Comment by rathsky** [comment]
Does seem like it's connected to offline messages, at least I noticed that the official steam client listed some during the last round of plugin kicks (meaning also, it obviously failed to retrieve them) and the latter stopped once I had viewed them in the client.

**Comment by rathsky** [comment]
Yes, the cause is most definitely that the plugin fails to retrieve offline messages and instead hits as 403. I've seen this happening consistently throughout the past days.

**Comment by ledti** [comment]
This started happening to me too, and it's definitely caused by offline messages.

**Comment by jgeboski** [comment]
This has been fixed as of 011375b.
